src/components/UserSelector.tsx

```diff
@@ -1,34 +1,27 @@
-import React, { useContext, useEffect, useState } from 'react';
+import React, { useEffect, useState } from 'react';
 import classNames from 'classnames';
-import { UserContext } from './UsersContext';
-import { User } from '../types/User';
+import { useAppDispatch, useAppSelector } from '../app/hooks';
+import { getUsers } from '../api/users';
+import { setUsers } from '../features/usersSlice';
+import { selectAuthor } from '../features/authorSlice';
 
-type Props = {
-  value: User | null;
-  onChange: (user: User) => void;
-};
-
-export const UserSelector: React.FC<Props> = ({
-  // `value` and `onChange` are traditional names for the form field
-  // `selectedUser` represents what actually stored here
-  value: selectedUser,
-  onChange,
-}) => {
-  // `users` are loaded from the API, so for the performance reasons
-  // we load them once in the `UsersContext` when the `App` is opened
-  // and now we can easily reuse the `UserSelector` in any form
-  const users = useContext(UserContext);
+export const UserSelector: React.FC = () => {
+  const users = useAppSelector(state => state.users);
+  const author = useAppSelector(state => state.author.selectedAuthor);
+  const dispatch = useAppDispatch();
   const [expanded, setExpanded] = useState(false);
 
+  useEffect(() => {
+    getUsers()
+      .then(response => dispatch(setUsers(response)));
```

Comment:
avoid mixing `.then().catch` and `async await` structures in one project. Rather chose one option and be consistent